modules/concurrency/include/hephaestus/concurrency/spinner.h

```diff
@@ -0,0 +1,34 @@
+//=================================================================================================
+// Copyright (C) 2023-2024 HEPHAESTUS Contributors
+//=================================================================================================
+
+#pragma once
+
+#include <atomic>
+#include <functional>
+#include <thread>
+
+namespace heph::concurrency {
+
+class Spinner {
+public:
+  Spinner();
+  virtual ~Spinner();
+  Spinner(const Spinner&) = delete;
+  auto operator=(const Spinner&) -> Spinner& = delete;
+  Spinner(Spinner&&) = delete;
+  auto operator=(Spinner&&) -> Spinner& = delete;
+
+  void start();
+  void stop();
+  virtual void spin();
```

Comment:
I think this should be private and non-virtual. 
If we want to add more logic we can do it and have a configurator that controls it.
For example, we could have 
```
struct SpinnerParams {
 std::chrono::duration period;
}
// ...

void spin() {
  while(!stop_requested_) {
    spinOnce();
    if (period > 0) { sleep_until(...); }|
  }
}
}
```
but in general I would say that we shouldn't let the user the option to control this

Reply:
fair enough, lets for now just make it private, if we need more control we can do once its needed.